internal/epoch/epoch_manager.go

```diff
@@ -950,6 +1023,101 @@ func (e *Manager) getCompleteIndexSetForCommittedState(ctx context.Context, cs C
 	return result, nil
 }
 
+type intRange struct {
+	lo, hi int
+}
+
+func (r intRange) length() int {
+	return r.hi - r.lo
+}
+
+func (r intRange) isEmpty() bool {
+	return r.length() <= 0
+}
+
+var errNonContiguousRange = errors.New("non-contiguous range")
+
+// Returns a continuous close-open epoch range for the keys, that is [lo, hi).
+// A range of the form [v,v) means the range is empty.
+// When the range is not continuous an error is returned.
+func getKeyRange[E any](m map[int]E) (intRange, error) {
+	keys := maps.Keys(m)
+
+	slices.Sort(keys)
+
+	keys = slices.Compact(keys)
```

Comment:
I think `slices.Compact` is redundant here, as keys are inherited from `map[int]E`, which by definition should not have duplicates.

Reply:
yeah, 🤦🏼